qiskit/assembler/assemble_schedules.py

```diff
@@ -82,6 +89,9 @@ def assemble_schedules(schedules, qobj_id, qobj_header, run_config):
             'instructions': qobj_instructions
         })
 
+    # set number of memoryslots
+    qobj_config['memory_slots'] = memory_slots or max_memory_slot
```

Comment:
Is it possible for `memory_slots` to be less than `max_memory_slot`? The spec says `memory_slots` will expand dynamically as the user uses them.

Reply:
This is tricky, I think. The question is do we want to do what the user says (set memory_slots) even if its stupid? Or should we dynamically increase the memory_slots? Or should we remove this argument period?

Reply:
I removed the ability to set memory slots, it will now be inferred from the schedule in a similar way to quantum circuit inferring this from the classical registers.